src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -0,0 +1,178 @@
+            throw ire("Source Keyspace '%s' doesn't exist", sourceKeyspace);
+
+        if (null == sourceTableMeta)
+            throw ire("Souce Table '%s'.'%s' doesn't exist", sourceKeyspace, sourceTableName);
+
+        if (sourceTableMeta.isIndex())
+            throw ire("Cannot use CREATE TABLE LIKE on a index table '%s'.'%s'.", sourceKeyspace, sourceTableName);
+
+        if (sourceTableMeta.isView())
+            throw ire("Cannot use CREATE TABLE LIKE on a materialized view '%s'.'%s'.", sourceKeyspace, sourceTableName);
+
+        KeyspaceMetadata targetKeyspaceMeta = schema.getNullable(targetKeyspace);
+        if (null == targetKeyspaceMeta)
+            throw ire("Target Keyspace '%s' doesn't exist", targetKeyspace);
+
+        if (targetKeyspaceMeta.hasTable(targetTableName))
+        {
+            if(ifNotExists)
+                return schema;
+
+            throw new AlreadyExistsException(targetKeyspace, targetTableName);
+        }
+        // todo support udt for differenet ks latter
+        if (!sourceKeyspace.equalsIgnoreCase(targetKeyspace) && !sourceKeyspaceMeta.types.isEmpty())
+            throw ire("Cannot use CREATE TABLE LIKE across different keyspace when source table have UDTs.");
+
+        String sourceCQLString = sourceTableMeta.toCqlString(false, false, true, false);
+        // add all user functions to be able to give a good error message to the user if the alter references
+        // a function from another keyspace
+        UserFunctions.Builder ufBuilder = UserFunctions.builder().add();
+        for (KeyspaceMetadata ksm : schema)
+            ufBuilder.add(ksm.userFunctions);
+
+        TableMetadata.Builder targetBuilder = CreateTableStatement.parse(sourceCQLString,
+                                                                         targetKeyspace,
+                                                                         targetTableName,
+                                                                         sourceKeyspaceMeta.types,
+                                                                         ufBuilder.build())
+                                                                  .indexes(Indexes.none())
+                                                                  .triggers(Triggers.none());
+
+        TableParams originalParams = targetBuilder.build().params;
+        TableParams newTableParams = attrs.asAlteredTableParams(originalParams);
+        TableMetadata table = targetBuilder.params(newTableParams)
+                                           .id(TableId.get(metadata))
+                                           .build();
+        table.validate();
+
+        if (targetKeyspaceMeta.replicationStrategy.hasTransientReplicas()
+            && table.params.readRepair != ReadRepairStrategy.NONE)
+        {
+            throw ire("read_repair must be set to 'NONE' for transiently replicated keyspaces");
+        }
+
+        if (!table.params.compression.isEnabled())
```

Comment:
So this says that: "if the compression is not enabled on the source table, then be sure that guardrail enables uncompressed sstables".

While it does make sense when creating the original table for the first time, does it still make sense to do when we are copying it?

If we disabled the compression on source table, then I _think_ that we already do have a guardrail for that, correct? But if we _change_ the guardrail in runtime after the original table was created (e.g. via JMX), then copying such table will not be possible anymore?

So, whether we can copy a table or not when the source table has disabled compression only depends on what the guardrail is set to in the runtime. 

I prefer to not check this. I would rather create a copy of the source table no matter what and we can propagate back to the client (the message will appear in cqlsh when copying table) that if guardrail is different, the a user might notified about that, but this is optional.

Reply:
Thinking about it again, I think how it is done is OK.